scripts/build-profile-gen.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+import sys
+import os
+import argparse
+import time
+import subprocess
+import json
+from datetime import datetime
+
+
+def parse_commandline():
+    if '--' in sys.argv:
+        idx = sys.argv.index('--')
+    else:
+        idx = 0
+
+    cmd = sys.argv[idx + 1:]
+
+    # attempt to determine output file
+    if '-o' in cmd:
+        output_path = cmd[cmd.index('-o') + 1]
+    else:
+        output_path = cmd[-1]
```

Comment:
This is very implicit. The way all our tools behave is that compilers use `-o` and linkers use thelast argument to specify the output file. You could do this more cleanly by having CMake pass a CLI arg to this script to let it know whether it is running a linkage or a compilation.